pkg/kube/podtargetresolver.go

```diff
@@ -53,9 +54,12 @@ func (o *PodTargetResolver) Close() {
 func (o *PodTargetResolver) List() []string {
 	pods := o.informer.GetStore().List()
 	var targets []string
+	var isPodEligible bool
+
 	for _, p := range pods {
 		pod := p.(*corev1.Pod)
-		if IsPodRunning(pod) && IsPodReady(pod) && pod.DeletionTimestamp == nil {
+		isPodEligible = o.skipStatusCheck || (IsPodRunning(pod) && IsPodReady(pod))
```

Comment:
I *think* that if skipStatusCheck is true, we still want to ensure IsPodRunning, but don't care about IsPodReady?